analytics/tests/test_counts.py

```diff
@@ -180,3 +180,66 @@ def test_active_users_by_is_bot(self):
         self.assertEqual(InstallationCount.objects.count(), 2)
         self.assertFalse(UserCount.objects.exists())
         self.assertFalse(StreamCount.objects.exists())
+
+    def test_messages_sent_by_message_type(self):
+        # type: () -> None
+        property = 'messages_sent:message_type'
+        stat = COUNT_STATS[property]
+
+        user1 = self.create_user('email1')
+        user2 = self.create_user('email2')
+        user3 = self.create_user('email3')
+
+        private_stream1, rec_private_stream1 = self.create_stream_with_recipient(name='private_stream1', invite_only=True)
+
+        public_stream1, rec_public_stream1 = self.create_stream_with_recipient(name='public_stream1')
+        public_stream2, rec_public_stream2 = self.create_stream_with_recipient(name='public_stream2')
+
+        # 2 private_stream
+        self.create_message(user1, rec_private_stream1)
+        self.create_message(user2, rec_private_stream1)
+
+        # 3 public_stream
+        self.create_message(user1, rec_public_stream1)
+        self.create_message(user2, rec_public_stream1)
+        self.create_message(user3, rec_public_stream1)
+
+        rec_user1 = Recipient.objects.create(type_id=user1.id, type=Recipient.PERSONAL)
+        rec_user2 = Recipient.objects.create(type_id=user2.id, type=Recipient.PERSONAL)
+        rec_user3 = Recipient.objects.create(type_id=user3.id, type=Recipient.PERSONAL)
+
+        huddle1, rec_huddle1 = self.create_huddle_with_recipient('foo')
+        huddle2, rec_huddle2 = self.create_huddle_with_recipient('bar')
+
+        # 5 private_message
+        self.create_message(user1, rec_user2)
+        self.create_message(user2, rec_user1)
+        self.create_message(user3, rec_user3)
+
+        self.create_message(user1, rec_huddle1)
+        self.create_message(user2, rec_huddle2)
+
+        do_fill_count_stat_at_hour(stat, self.TIME_ZERO)
+
+        self.assertCountEquals(RealmCount, property, 2, subgroup='private_stream', interval=stat.interval)
+        self.assertCountEquals(RealmCount, property, 3, subgroup='public_stream', interval=stat.interval)
+        self.assertCountEquals(RealmCount, property, 5, subgroup='private_message', interval=stat.interval)
+
+    def test_messages_sent_by_message_type_same_id(self):
+        # type: () -> None
+        property = 'messages_sent:message_type'
+        stat = COUNT_STATS[property]
+
+        user = self.create_user('email', id=200)
```

Comment:
Small thing, but let's use 1000 (or a different favorite number) instead of 200, since 200 has a specific meaning (the http response), and so it may show up in people's `git grep`ing through the codebase.